zql/zql.peg

```diff
@@ -37,25 +41,29 @@
 #define ISNOTNULL(n) (n)
 #define ASSERT_ARRAY(a) a
 #define ASSERT_STRING(s) s
+#define ASSERT_MAP(m) m
+#define HAS(m, k) (k in m)
+#define MAP(...) {__VA_ARGS__}
+#define PRINT(...) console.log(__VAR_ARGS__)
 #endif
 
 
 start = _? ast:query _? EOF { RETURN(ast) }
 
 query
   = procs:procChain {
-      INIT_ASSIGN_VAR(filt, makeFilterProc(makeMatchAll()))
-      RETURN(makeSequentialProc(PREPEND(filt, procs)))
+      VAR(filt) = MAP("op": "FilterProc", "filter": MAP("op": "MatchAll"))
+	  RETURN(MAP("op": "SequentialProc", "procs": PREPEND(filt, procs)))
```

Comment:
The indentation here is inconsistent, as it is in a number of other places in this file.

Reply:
quite right